### PR TITLE
removed setter for GRPCService eval field

### PR DESCRIPTION
**Pull request by james-milligan**
Removed the SetEval method. Moved unit tests into package to allow for field access without setter function.

Signed-off-by: James-Milligan <james@omnant.co.uk>

### CODE REVIEW

File: pkg/service/grpc_service_test.go
Comment (toddbaert) [inline pkg/service/grpc_service_test.go:1]: Any opinions on tests being a separate package?

On the one hand, the same package might make things easier... on the other, using a different package forces some "blackbox" testing since we lose access to the packages private identifiers.

Just thinking aloud...

**Review by toddbaert — APPROVED** [review]
LGTM, I think this resolves @AlexsJones 's concern... He's on PTO so I'll leave it up to you to choose to merge or not without his input.

File: pkg/service/grpc_service.go
Comment (AlexsJones) [inline pkg/service/grpc_service.go:29]: I never got time to finish reviewing before this got merged, why is the assignment being done this way?

File: pkg/service/grpc_service_test.go
Comment (AlexsJones) [reply pkg/service/grpc_service_test.go:1]: It's not just that, it also gets access to private variables

File: pkg/service/grpc_service.go
Comment (james-milligan) [reply pkg/service/grpc_service.go:29]: Assignment was previously being handled within the Serve method, for putting together tests access was needed to set a mocked eval. 
As per your comment I have moved the tests into the same package to allow access to the un-exported field, preventing the need for a setter, alternatively we can make eval available as an exported field and put the tests back in their own package.

**Comment by AlexsJones** [comment]
Thanks this is far more idiomatic and readable

**Review by AlexsJones — APPROVED** [review]

**Pull request merged**
